src/schema/v2/notifications/index.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  GraphQLBoolean,
+  GraphQLFieldConfig,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { pageable } from "relay-cursor-paging"
+import {
+  connectionWithCursorInfo,
+  createPageCursors,
+} from "schema/v2/fields/pagination"
+import { ResolverContext } from "types/graphql"
+import { IDFields, NodeInterface } from "../object_identification"
+import { date } from "schema/v2/fields/date"
+import { artworkConnection } from "../artwork"
+import { connectionFromArray, connectionFromArraySlice } from "graphql-relay"
+import numeral from "../fields/numeral"
+import { pick } from "lodash"
+
+export const NotificationType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Notification",
+  interfaces: [NodeInterface],
+  fields: () => ({
+    ...IDFields,
+    title: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: ({ actors }) => actors,
+    },
+    message: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: ({ message }) => message.toLowerCase(),
+    },
+    isUnread: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+      resolve: ({ status }) => status === "unread",
+    },
+    publishedAt: date(({ date }) => date),
+    href: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: ({ object }) => `/artist/${object.artist.id}/works-for-sale`,
+    },
+    artworksConnection: {
+      type: artworkConnection.connectionType,
+      args: pageable(),
+      resolve: async ({ object_ids: ids }, args, { artworksLoader }) => {
+        const { page, size } = convertConnectionArgsToGravityArgs(args)
+        return artworksLoader({ ids, batched: true }).then((body) => {
```

Comment:
I haven't seen `batched` in use before — what does this do for us?

Reply:
There's some context in this PR: https://github.com/artsy/metaphysics/pull/2537

and it might make sense to use a similarly named variable here to better communicate the intent. The gist is that by passing `batched: true` we're instructing the data loader to pass the id arguments in the exact order that we're specifying here rather than sorting the params to increase the likelihood of cache hits.

Reply:
Got it, thanks!